pkg/ccl/testccl/sqlccl/sqlccl_test: TestExplainRedactDDL failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainRedactDDL [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12099312?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12099312?buildTab=artifacts#/) on master @ [76a8d1077d3d29bcf020a48972efd1f77aa892c2](https://github.com/cockroachdb/cockroach/commits/76a8d1077d3d29bcf020a48972efd1f77aa892c2):


```
=== RUN   TestExplainRedactDDL/Explain/NoMode/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Catalog (0.01s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Memo (0.01s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Shape
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Shape (0.01s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Viz (0.03s)
=== RUN   TestExplainRedactDDL/Explain/Opt/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/Catalog (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Vec/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/NoFlag (0.01s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/Memo (0.02s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/NoFlag (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Opt/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/NoFlag (0.01s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/Verbose
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/Verbose (0.02s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog (0.02s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/Shape
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Shape (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Viz (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Vec/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/Memo (0.01s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze
    --- FAIL: TestExplainRedactDDL/ExplainAnalyze (0.46s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/Viz (0.02s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/NoFlag (0.02s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedactDDL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32187

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainRedactDDL [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12099354?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12099354?buildTab=artifacts#/) on master @ [76a8d1077d3d29bcf020a48972efd1f77aa892c2](https://github.com/cockroachdb/cockroach/commits/76a8d1077d3d29bcf020a48972efd1f77aa892c2):


```
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Verbose (0.13s)
=== RUN   TestExplainRedactDDL/Explain
    --- FAIL: TestExplainRedactDDL/Explain (4.44s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Viz (0.22s)
=== RUN   TestExplainRedactDDL/Explain/Opt
        --- FAIL: TestExplainRedactDDL/Explain/Opt (0.57s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Verbose
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Verbose (0.17s)
=== RUN   TestExplainRedactDDL/Explain/Plan
        --- FAIL: TestExplainRedactDDL/Explain/Plan (1.13s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Shape
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Shape (0.15s)
=== RUN   TestExplainRedactDDL/Explain/Vec/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/Viz (0.07s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze
    --- FAIL: TestExplainRedactDDL/ExplainAnalyze (2.96s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/NoFlag (0.13s)
=== RUN   TestExplainRedactDDL/Explain/Vec/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/Catalog (0.06s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/NoFlag (0.12s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan
        --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan (0.84s)
=== RUN   TestExplainRedactDDL/Explain/NoMode
        --- FAIL: TestExplainRedactDDL/Explain/NoMode (1.41s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Memo (0.14s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Shape
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Shape (0.19s)
=== RUN   TestExplainRedactDDL/Explain/Opt/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/Memo (0.08s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Catalog (0.13s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog (0.11s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedactDDL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestExplainRedactDDL [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12107739?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12107739?buildTab=artifacts#/) on master @ [330500d0433bae42bc7c0d6842a427aab3b27f03](https://github.com/cockroachdb/cockroach/commits/330500d0433bae42bc7c0d6842a427aab3b27f03):


```
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/NoFlag (0.03s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Memo (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Opt
        --- FAIL: TestExplainRedactDDL/Explain/Opt (0.11s)
=== RUN   TestExplainRedactDDL/Explain/Plan
        --- FAIL: TestExplainRedactDDL/Explain/Plan (0.12s)
=== RUN   TestExplainRedactDDL/Explain/Vec/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/Memo (0.01s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/Memo (0.02s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Catalog (0.03s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/Plan/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/Plan/Memo (0.02s)
=== RUN   TestExplainRedactDDL/Explain/NoMode/Shape
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/NoMode/Shape (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Memo
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Memo (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Viz (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Plan/NoFlag
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/NoFlag (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Plan/Verbose
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Plan/Verbose (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Vec/Types
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Vec/Types (0.01s)
=== RUN   TestExplainRedactDDL/ExplainAnalyze/NoMode/Types
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/ExplainAnalyze/NoMode/Types (0.02s)
=== RUN   TestExplainRedactDDL/Explain/Opt/Catalog
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/Catalog (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Opt/Types
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/Types (0.01s)
=== RUN   TestExplainRedactDDL/Explain/Opt/Viz
    explain_test_util.go:97: pq: internal error: comparison overload not found (is, refcursor, unknown)
            --- FAIL: TestExplainRedactDDL/Explain/Opt/Viz (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedactDDL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/docs/generated/sql/operators.md
+++ b/docs/generated/sql/operators.md
@@ -447,6 +447,7 @@
 <tr><td>oid <code>IS NOT DISTINCT FROM</code> <a href="int.html">int</a></td><td><a href="bool.html">bool</a></td></tr>
 <tr><td>oid <code>IS NOT DISTINCT FROM</code> oid</td><td><a href="bool.html">bool</a></td></tr>
 <tr><td>pg_lsn <code>IS NOT DISTINCT FROM</code> pg_lsn</td><td><a href="bool.html">bool</a></td></tr>
+<tr><td>refcursor <code>IS NOT DISTINCT FROM</code> unknown</td><td><a href="bool.html">bool</a></td></tr>
 <tr><td><a href="string.html">string</a> <code>IS NOT DISTINCT FROM</code> <a href="string.html">string</a></td><td><a href="bool.html">bool</a></td></tr>
 <tr><td><a href="string.html">string[]</a> <code>IS NOT DISTINCT FROM</code> <a href="string.html">string[]</a></td><td><a href="bool.html">bool</a></td></tr>
 <tr><td><a href="time.html">time</a> <code>IS NOT DISTINCT FROM</code> <a href="time.html">time</a></td><td><a href="bool.html">bool</a></td></tr>

--- a/pkg/sql/logictest/testdata/logic_test/refcursor
+++ b/pkg/sql/logictest/testdata/logic_test/refcursor
@@ -440,13 +440,6 @@ SELECT 'foo'::REFCURSOR >= 'foo'::TEXT;
 statement error pgcode 22023 pq: unsupported comparison operator
 SELECT 'foo'::REFCURSOR >= NULL;
 
-# TODO(drewk): Postgres allows this case.
-statement error pgcode 42883 pq: unsupported comparison operator
-SELECT 'foo'::REFCURSOR IS NULL;
-
-statement error pgcode 42883 pq: unsupported comparison operator
-SELECT 'foo'::REFCURSOR IS NOT NULL;
-
 statement error pgcode 22023 pq: unsupported comparison operator
 SELECT 'foo'::REFCURSOR IS DISTINCT FROM 'foo'::REFCURSOR;
 
@@ -454,15 +447,110 @@ statement error pgcode 22023 pq: unsupported comparison operator
 SELECT 'foo'::REFCURSOR IS DISTINCT FROM 'foo'::TEXT;
 
 statement error pgcode 22023 pq: unsupported comparison operator
-SELECT 'foo'::REFCURSOR IS DISTINCT FROM NULL;
-
-statement error pgcode 22023 pq: unsupported comparison operator
 SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM 'foo'::REFCURSOR;
 
 statement error pgcode 22023 pq: unsupported comparison operator
 SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM 'foo'::TEXT;
 
-statement error pgcode 22023 pq: unsupported comparison operator
+# Regression test for #112010 - REFCURSOR should allow IS NULL and
+# IS NOT NULL comparisons.
+subtest is_null
+
+query B
+SELECT 'foo'::REFCURSOR IS NULL;
+----
+false
+
+query B
+SELECT 'foo'::REFCURSOR IS NOT NULL;
+----
+true
+
+query B
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM NULL;
+----
+true
+
+query B
 SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM NULL;
+----
+false
+
+# REFCURSOR columns are allowed.
+query B rowsort
+SELECT l IS NULL FROM implicit_types;
+----
+true
+true
+true
+true
+true
+
+query B rowsort
+SELECT l IS NOT NULL FROM implicit_types;
+----
+false
+false
+false
+false
+false
+
+query B rowsort
+SELECT l IS DISTINCT FROM NULL FROM implicit_types;
+----
+false
+false
+false
+false
+false
+
+query B rowsort
+SELECT l IS NOT DISTINCT FROM NULL FROM implicit_types;
+----
+true
+true
+true
+true
+true
+
+# Comparison with column is not allowed.
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM l FROM implicit_types;
+
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM l FROM implicit_types;
+
+# Comparison with typed NULL is not allowed.
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM (NULL::REFCURSOR);
+
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM (NULL::REFCURSOR);
+
+# Comparison with CASE expression is not allowed.
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM (CASE WHEN true THEN NULL ELSE 1 END);
+
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM (CASE WHEN true THEN NULL ELSE 1 END);
+
+# The CASE operator is folded into an untyped NULL, so it's allowed.
+# TODO(drewk): Postgres doesn't allow this case.
+query B
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM (CASE WHEN true THEN NULL ELSE NULL END);
+----
+true
+
+query B
+SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM (CASE WHEN true THEN NULL ELSE NULL END);
+----
+false
+
+# The CASE operator is folded into a typed NULL, so it's not allowed.
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS DISTINCT FROM (CASE WHEN true THEN NULL ELSE NULL::REFCURSOR END);
+
+statement error pgcode 22023 pq: unsupported comparison operator
+SELECT 'foo'::REFCURSOR IS NOT DISTINCT FROM (CASE WHEN true THEN NULL ELSE NULL::REFCURSOR END);
 
 subtest end

--- a/pkg/sql/sem/tree/eval.go
+++ b/pkg/sql/sem/tree/eval.go
@@ -1775,6 +1775,10 @@ var CmpOps = cmpOpFixups(map[treecmp.ComparisonOperatorSymbol]*CmpOpOverloads{
 		makeIsFn(types.Void, types.Unknown, volatility.Leakproof),
 		makeIsFn(types.Unknown, types.Void, volatility.Leakproof),
 
+		// REFCURSOR cannot be compared with itself, but as a special case, it can
+		// be compared with NULL using IS NOT DISTINCT FROM.
+		makeIsFn(types.RefCursor, types.Unknown, volatility.Leakproof),
+
 		// Tuple comparison.
 		{
 			LeftType:          types.AnyTuple,

--- a/pkg/sql/sem/tree/type_check.go
+++ b/pkg/sql/sem/tree/type_check.go
@@ -941,14 +941,6 @@ func (expr *ComparisonExpr) TypeCheck(
 		return nil, err
 	}
 
-	// REFCURSOR does not support comparisons.
-	leftType, rightType := leftTyped.ResolvedType(), rightTyped.ResolvedType()
-	if leftType.Family() == types.RefCursorFamily || rightType.Family() == types.RefCursorFamily {
-		return nil, pgerror.Newf(pgcode.UndefinedFunction,
-			"unsupported comparison operator: <%s> %s <%s>", leftType, cmpOp, rightType,
-		)
-	}
-
 	if alwaysNull {
 		return DNull, nil
 	}
@@ -1471,12 +1463,6 @@ func (expr *IsNullExpr) TypeCheck(
 	if err != nil {
 		return nil, err
 	}
-	// REFCURSOR does not support comparisons.
-	if exprTyped.ResolvedType().Family() == types.RefCursorFamily {
-		return nil, pgerror.New(pgcode.UndefinedFunction,
-			"unsupported comparison operator: refcursor IS unknown",
-		)
-	}
 	expr.Expr = exprTyped
 	expr.typ = types.Bool
 	return expr, nil
@@ -1489,12 +1475,6 @@ func (expr *IsNotNullExpr) TypeCheck(
 	exprTyped, err := expr.Expr.TypeCheck(ctx, semaCtx, types.Any)
 	if err != nil {
 		return nil, err
-	}
-	// REFCURSOR does not support comparisons.
-	if exprTyped.ResolvedType().Family() == types.RefCursorFamily {
-		return nil, pgerror.New(pgcode.UndefinedFunction,
-			"unsupported comparison operator: refcursor IS NOT unknown",
-		)
 	}
 	expr.Expr = exprTyped
 	expr.typ = types.Bool

